README.md
## Description

Installs/Configures [Atlassian Crowd](https://www.atlassian.com/software/crowd/) server.

## Requirements

### Platforms

* CentOS 6
* RHEL 6
* Ubuntu 12.04, 12.10, 13.04

### Chef

* The cookbook requires Chef 11 due to `ark` usage

### Databases

* HSQLDB (not recommended for production usage)
* MySQL
* Postgres
* Percona

### Cookbooks

Required [Opscode Cookbooks](https://github.com/opscode-cookbooks/)

* [apache2](https://github.com/opscode-cookbooks/apache2) (if using Apache 2 proxy)
* [ark](https://github.com/opscode-cookbooks/ark)
* [database](https://github.com/opscode-cookbooks/database)
* [java](https://github.com/opscode-cookbooks/java)
* [mysql](https://github.com/opscode-cookbooks/mysql) (if using MySQL database)
* [postgresql](https://github.com/opscode-cookbooks/postgresql) (if using Postgres database)

Third-Party Cookbooks

* [percona](https://github.com/phlipper/chef-percona) (if using Percona database)

## Attributes

These attributes are under the `node['crowd']` namespace.

Attribute | Description | Type | Default
----------|-------------|------|--------
checksum | SHA256 checksum for Crowd install | String | auto-detected (see attributes/default.rb)
home_path | home data directory for Crowd user | String | /var/atlassian/application-data/crowd
install_path | location to install Crowd | String | /opt/atlassian
install_type | Crowd install type - "standalone" only for now | String | standalone
url_base | URL base for Crowd install | String | http://www.atlassian.com/software/crowd/downloads/binary/atlassian-crowd
url | URL for Crowd install | String | auto-detected (see attributes/default.rb)
user | user to run Crowd | String | crowd
version | Crowd version to install | String | 2.7.2

### Crowd Database Attributes

All of these `node['crowd']['database']` attributes are overridden by `crowd/crowd` encrypted data bag (Hosted Chef) or data bag (Chef Solo), if it exists

Attribute | Description | Type | Default
----------|-------------|------|--------
host | FQDN or "localhost" (localhost automatically installs `['database']['type']` server) | String | localhost
name | Crowd database name | String | crowd
password | Crowd database user password | String | changeit
port | Crowd database port | Fixnum | 3306
testInterval | Crowd database pool idle test interval in minutes | Fixnum | 2
type | Crowd database type - "hsqldb" (not recommended), "mysql", "percona" or "postgresql" | String | mysql
user | Crowd database user | String | crowd

### Crowd JVM Attributes

These attributes are under the `node['crowd']['jvm']` namespace.

Attribute | Description | Type | Default
----------|-------------|------|--------
minimum_memory | JVM minimum memory | String | 512m
maximum_memory | JVM maximum memory | String | 768m
maximum_permgen | JVM maximum PermGen memory | String | 256m
java_opts | additional JAVA_OPTS to be passed to Crowd JVM during startup | String | ""
support_args | additional JAVA_OPTS recommended by Atlassian support for Crowd JVM during startup | String | ""

### Crowd Tomcat Attributes

These attributes are under the `node['crowd']['tomcat']` namespace.

Any `node['crowd']['tomcat']['key*']` attributes are overridden by `crowd/crowd` encrypted data bag (Hosted Chef) or data bag (Chef Solo), if it exists

Attribute | Description | Type | Default
----------|-------------|------|--------
keyAlias | Tomcat SSL keystore alias | String | tomcat
keystoreFile | Tomcat SSL keystore file - will automatically generate self-signed keystore file if left as default | String | `#{node['crowd']['home_path']}/.keystore`
keystorePass | Tomcat SSL keystore passphrase | String | changeit
port | Tomcat HTTP port | Fixnum | 8095
ssl_port | Tomcat HTTPS port | Fixnum | 8443
context | Tomcat context | String | ""

## Recipes

* `recipe[crowd]` Installs Atlassian Crowd with built-in Tomcat and Apache 2 proxy
* `recipe[crowd::apache2]` Installs/configures Apache 2 proxy for Crowd (ports 80/443)
* `recipe[crowd::configuration]` Configures Crowd's settings
* `recipe[crowd::database]` Installs/configures MySQL/Percona/Postgres server, database, and user for Crowd
* `recipe[crowd::linux_standalone]` Installs/configures Crowd via Linux standalone archive
* `recipe[crowd::service_init]` Installs/configures Crowd init service
* `recipe[crowd::tomcat_configuration]` Configures Crowd's built-in Tomcat

## Contributing

Please use standard Github issues/pull requests.

## License and Contributors

Please see license information in: [LICENSE](LICENSE)

* Sergio Leone <sergio.leone@qvc.com>


CHANGELOG.md
# 0.2.0

* Enhancements: Default to Crowd 2.8.0

# 0.1.0

Initial release of chef-crowd

* Enhancements
  * an enhancement

* Bug Fixes
  * a bug fix


attributes/default.rb
default['crowd']['home_path']    = '/var/atlassian/application-data/crowd'
default['crowd']['install_path'] = '/opt/atlassian'
default['crowd']['install_type'] = 'standalone'
default['crowd']['service_type'] = 'init'
default['crowd']['url_base']     = 'http://www.atlassian.com/software/crowd/downloads/binary/atlassian-crowd'
default['crowd']['user']         = 'crowd'
default['crowd']['version']      = '2.8.0'

default['crowd']['url']      = "#{node['crowd']['url_base']}-#{node['crowd']['version']}.tar.gz"
default['crowd']['checksum'] =
case node['crowd']['version']
when '2.7.2' then '49361f2c7cbd8035c2fc64dfff098eb5e51d754b5645425770da14fc577f1048'
when '2.8.0' then 'c857eb16f65ed99ab8b289fe671e3cea89140d42f85639304caa91a3ba9ade05'
end

default['crowd']['apache2']['webapp_enable']      = true
default['crowd']['apache2']['access_log']         = ''
default['crowd']['apache2']['error_log']          = ''
default['crowd']['apache2']['port']               = 80
default['crowd']['apache2']['virtual_host_alias'] = node['fqdn']
default['crowd']['apache2']['virtual_host_name']  = node['hostname']

default['crowd']['apache2']['ssl']['access_log']       = ''
default['crowd']['apache2']['ssl']['chain_file']       = ''
default['crowd']['apache2']['ssl']['error_log']        = ''
default['crowd']['apache2']['ssl']['port']             = 443

case node['platform_family']
when 'rhel'
  default['crowd']['apache2']['ssl']['certificate_file'] = '/etc/pki/tls/certs/localhost.crt'
  default['crowd']['apache2']['ssl']['key_file']         = '/etc/pki/tls/private/localhost.key'
else
  default['crowd']['apache2']['ssl']['certificate_file'] = '/etc/ssl/certs/ssl-cert-snakeoil.pem'
  default['crowd']['apache2']['ssl']['key_file']         = '/etc/ssl/private/ssl-cert-snakeoil.key'
end

default['crowd']['database']['host']     = 'localhost'
default['crowd']['database']['name']     = 'crowd'
default['crowd']['database']['password'] = 'changeit'
default['crowd']['database']['port']     = 3306
default['crowd']['database']['testInterval'] = 2
default['crowd']['database']['type']     = 'mysql'
default['crowd']['database']['user']     = 'crowd'

default['crowd']['mysql_connector']['version']  = '5.1.31'
default['crowd']['mysql_connector']['checksum'] = '5a253a2aa8f06758c1901c3862402c69db601e9de456864ab1922955b225bba6'

default['crowd']['jvm']['minimum_memory']  = '512m'
default['crowd']['jvm']['maximum_memory']  = '768m'
default['crowd']['jvm']['maximum_permgen'] = '256m'
default['crowd']['jvm']['java_opts']       = ''
default['crowd']['jvm']['support_args']    = ''

default['crowd']['tomcat']['keyAlias']     = 'tomcat'
default['crowd']['tomcat']['keystoreFile'] = "#{node['crowd']['home_path']}/.keystore"
default['crowd']['tomcat']['keystorePass'] = 'changeit'
default['crowd']['tomcat']['port']         = '8095'
default['crowd']['tomcat']['ssl_port']     = '8443'
default['crowd']['context']                = 'crowd'
